firewood/src/merkle/parallel.rs

```diff
@@ -227,8 +227,37 @@ impl ParallelMerkle {
                     }
                     // Sent from the coordinator to the workers to signal that the batch is done.
                     Request::Done => {
+                        // Hash this subtrie and return the root as a Child::MaybePersisted.
+                        let hashed_result = merkle
+                            .nodestore
+                            .root_mut()
+                            .take()
+                            .map(|root| {
+                                #[cfg(not(feature = "ethhash"))]
+                                let (root_node, root_hash) =
+                                    NodeStore::<MutableProposal, FileBacked>::hash_helper(
+                                        root,
+                                        Path::from_nibbles_iterator(once(first_nibble)),
```

Comment:
I think you can use the `impl<T: AsRef<[u8]>> From<T> for Path` instead of making an interator just to construct the path (untested):
```suggestion
                                        Path::from(&[first_nibble]),
```

(also in the ethhash case)